user.py

```diff
@@ -0,0 +1,360 @@
+
+    def sanitise_phone(self, phone_number):
+        if phone_number != "":
+            phone_number = re.sub(r"[^0-9]", "", phone_number)
+            if phone_number.startswith("0"):
+                phone_number = "+44" + phone_number[1:]
+            if phone_number.startswith("44"):
+                phone_number = "+44" + phone_number[2:]
+            return phone_number
+        return ""
+
+    def sanitise_name(self, name):
+        return re.sub(r"[^a-zA-Z0-9-_\']", "", name)
+
+    def update(self, name, phone_number, custom_paths, is_la, group):
+        if self.get_details() == {}:
+            return False
+
+        if (
+            name is None
+            and phone_number is None
+            and custom_paths is None
+            and is_la is None
+            and group is None
+        ):
+            return False
+        attrs = []
+        try:
+            attrs += self.__attribute("custom:is_la", is_la)
+            attrs += self.__attribute("name", self.sanitise_name(name))
+            attrs += self.__custom_path_attribute(is_la, custom_paths, group)
+            attrs += self.__phone_number_attribute(phone_number)
+            self.set_group(group)
+        except ValueError:
+            return False
+        user_attributes = {}
+        user_attributes["UserAttributes"] = attrs
+        return make_request(
+            "admin_update_user_attributes", self.email_address, user_attributes,
+        )
+
+    def __attribute(self, field_name, value):
+        if value is None:
+            return []
+
+        if isinstance(value, str):
+            return [{"Name": field_name, "Value": value}]
+        else:
+            raise ValueError(f"{value}: is not expected value for field: {field_name}")
+
+    def __custom_path_attribute(self, is_la, custom_paths, group_name):
+        if custom_paths is None:
+            return []
+
+        if isinstance(custom_paths, str):
+            group_name = (
+                group_name if group_name is not None else self.details["group"]["value"]
+            )
+            if not self.user_paths_are_valid(is_la, custom_paths, group_name):
+                raise ValueError(f"custom paths: is not expected value")
+
+            return [{"Name": "custom:paths", "Value": custom_paths}]
+        else:
+            raise ValueError(f"custom paths: is not expected value")
+
+    def user_paths_are_valid(self, is_la, paths_semicolon_separated, group_name):
+        if "admin" not in group_name and paths_semicolon_separated == "":
+            return False
+
+        app_authorised_paths = [os.getenv("BUCKET_MAIN_PREFIX", "web-app-prod-data")]
+        for authorised_path in app_authorised_paths:
+            for path in paths_semicolon_separated.split(";"):
+                la_path = "{}/local_authority/".format(authorised_path)
+                # if new attr for is_la is 0 (not a local authority)
+                # then don't allow local local_authority paths to be set
```

Comment:
What does "don't allow local local_authority paths" mean? I'm trying to figure out if the first "local" in that sentence is a typo or refers to a local variable 🤔 .

Reply:
I copied it and pasted it without much thinking about it tbh, I think it was a typo. I've fixed that now

Reply:
No worries! I wondered if that was the case. Thanks for fixing.